diff -N -a -u -r StormLib.original/makefile.w32 StormLib/makefile.w32
--- StormLib.original/makefile.w32	2010-09-08 08:20:40 -0400
+++ StormLib/makefile.w32	2010-10-18 02:21:12 -0400
@@ -8,15 +8,19 @@
 #    
 #####################################################################
 
-PATH   = C:\TOOLS32\MINGW\bin
-CPP    = g++
-CC     = gcc
+#PATH   = C:\TOOLS32\MINGW\bin
+MINGW = i686-w64-mingw32-
+#MINGW = x86_64-w64-mingw32-
+CPP    = $(MINGW)g++
+CC     = $(MINGW)gcc
 CFLAGS = -Wall
+#CFLAGS += -ggdb
+CFLAGS += -O
 DFLAGS = -D_7ZIP_ST
 LFLAGS = -shared
-#LFLAGS = -lbz2 -lz
-AR     = ar
-ARFLAGS= rcs
+#LFLAGS += -lbz2 -lz
+AR     = $(MINGW)ar
+ARFLAGS= $(MINGW)rcs
 
 LIBRARY  = StormLib.dll
 #LIBRARY  = StormLib.lib
@@ -25,7 +29,8 @@
            src/huffman/huff.o \
            src/sparse/sparse.o \
            src/FileStream.o \
-           src/SCommon.o \
+           src/SBaseCommon.o \
+           src/SBaseFileTable.o \
            src/SCompression.o \
            src/SFileAddFile.o \
            src/SFileAttributes.o \
@@ -246,8 +251,8 @@
 
 all: $(LIBRARY)
 
-#clean:
-#   rm -f $(OBJS) $(LIBRARY)
+clean:
+	rm -f $(OBJS_C) $(OBJS_CPP) $(LIBRARY)
 
 #install: $(LIBRARY)
 #   install $(LIBRARY) /usr/local/lib
diff -N -a -u -r StormLib.original/src/SFileExtractFile.cpp StormLib/src/SFileExtractFile.cpp
--- StormLib.original/src/SFileExtractFile.cpp	2010-09-06 04:19:46 -0400
+++ StormLib/src/SFileExtractFile.cpp	2010-10-18 01:46:08 -0400
@@ -12,7 +12,7 @@
 #include "StormLib.h"
 #include "StormCommon.h"
 
-bool WINAPI SFileExtractFile(HANDLE hMpq, const char * szToExtract, const char * szExtracted)
+bool WINAPI SFileExtractFile(HANDLE hMpq, const char * szToExtract, const char * szExtracted, bool openPatched)
 {
     TFileStream * pLocalFile = NULL;
     HANDLE hMpqFile = NULL;
@@ -21,7 +21,7 @@
     // Open the MPQ file
     if(nError == ERROR_SUCCESS)
     {
-        if(!SFileOpenFileEx(hMpq, szToExtract, SFILE_OPEN_FROM_MPQ, &hMpqFile))
+        if(!SFileOpenFileEx(hMpq, szToExtract, (openPatched ? SFILE_OPEN_PATCHED_FILE: SFILE_OPEN_FROM_MPQ), &hMpqFile))
             nError = GetLastError();
     }
 
diff -N -a -u -r StormLib.original/src/StormLib.h StormLib/src/StormLib.h
--- StormLib.original/src/StormLib.h	2010-09-29 15:04:28 -0400
+++ StormLib/src/StormLib.h	2010-10-18 01:39:48 -0400
@@ -790,7 +790,7 @@
 bool   WINAPI SFileGetFileInfo(HANDLE hMpqOrFile, DWORD dwInfoType, void * pvFileInfo, DWORD cbFileInfo, LPDWORD pcbLengthNeeded = NULL);
 
 // High-level extract function
-bool   WINAPI SFileExtractFile(HANDLE hMpq, const char * szToExtract, const char * szExtracted);
+bool   WINAPI SFileExtractFile(HANDLE hMpq, const char * szToExtract, const char * szExtracted, bool openPatched = true);
 
 //-----------------------------------------------------------------------------
 // Functions for file and archive verification
